### DISCUSSION TITLE
Page Tree Metadata

### DISCUSSION THREAD

**Discussion by lehoule** [WikiTransformationProject/wikitraccs-wikipakk-releases#26] [Q&A] [answered]
I'm currently doing a POC to convert our Confluence to Sharepoint. 

My main pain point for now is the page tree. Your wikipakk solution looks amazing. I didn't test it yet because I would need to go through our approval process to add it. Before I do that, I have a few questions:

First, when I migrate my pages from Confluence I see that the pages have properties that I guess let's you reconstruct the tree: Confluence: Parent Id (WikiTraccs) and Confluence: Sibling Order (WikiTraccs). Once the plugin is installed, do you copy the values in new properties so it's less reliant on old data or it's only with new pages?

Also, do you have any guarantee of support or open/private sourcing the code if one day you no longer wish to maintain it? We have over 25000 pages in our wiki. It would be a lot of trouble to redo the page trees again!

Thanks! 
Your documentation is amazing by the way!

**Comment by heinrich-ulbricht** [comment, marked as answer]
@lehoule Hi, thanks for trying out WikiTraccs and WikiPakk, and for the positive feedback!

The WikiPakk page tree indeed looks at three fields in the Site Pages library—Confluence Id, Confluence Parent Id, and Confluence Sibling Order. These are populated by WikiTraccs when it migrates pages from Confluence to SharePoint. If the page tree detects those values, it uses them to rebuild the hierarchy for migrated pages.

For newly created pages, the hierarchy is stored in a hidden hierarchy list named “DONOTMODIFY WikiTraccs Page Tree Data.” The page tree checks both the Site Pages library (for migrated pages) and this hierarchy list (for newly created pages) to produce the final structure.

While not common, it’s technically possible to migrate hierarchy data from the Site Pages library to the hierarchy list. Here’s a sample PowerShell script that demonstrates how to manipulate the hierarchy list: https://github.com/WikiTransformationProject/library/blob/main/scripts/wikipakk/create-hierarchy/CreateHierarchy.ps1

Regarding maintenance and future support: There isn’t a formal maintenance guarantee or a publicly available source code release policy. I understand that this may factor into your decision.

Thanks again for getting in touch. I hope this helps, and feel free to reach out if you have more questions!

**Reply by lehoule** [reply]
Thank you for your answer. We will be able to do a well-informed decision.
